### ISSUE TITLE
MedicationStatement:dosageLastChanged

### ISSUE THREAD

**Issue by artronics** [NHSDigital/fhir-transforms#40]
We dont' have any examples related to this extensions. This ticket [AMB-1022](https://nhsd-jira.digital.nhs.uk/browse/AMB-1022) captures the requirement and it's currently blocked.
The documentation says:
> MedicationDosageLastChanged extension does not currently map to UKCore but is under review

**Comment by declankieran** [comment]
@artronics Can I get access to your Jira please?  Also probably worth tagging me when you open these, so I get a notification.

**Comment by declankieran** [comment]
@artronics sorry just noticed I wasn't watching the repo, sorted now, I think I should be notified on all updates now

**Comment by artronics** [comment]
@declankieran-nhsd Any update regarding this extension?. In the description it says "it's under review"

**Comment by declankieran** [comment]
No, but I'll also bring this up in the DA on Tuesday.  I'll also have a look over it myself today.

**Comment by declankieran** [comment]
The only note I've found in the design documented so far is

12/09/2019 - Dosage.text warning (to include the date the dosage was last changed) and for the Complex Extension with the "dosage last change date".

Dosage is embedded in the MedicationStatement as opposed to just being a reference.  If it was a reference you would have the metadata of all linked dosage resources to use to indicate when it was last changed, but I'm guessing its maybe not a reference for clinical safety around it then being possible to modify it outside of the MedicationStatement and having no notion of whether that was intentional in the context of the MedicationStatement.  It being embedded into the MedicationStatement resource means it's bound within the version history of the MedicationStatement resource.

If the dosage is changed and that needed to be determined though, it could be determined by examining the version history of MedicationStatement resource.  The requirement above states that the Dosage.text should include "the date the dosage was last changed".  This could be done by the application writing the resource, but it would be difficult to validate that without the use of a well defined datatype, i.e. the extension including the value.  If it was something to validate on though, I think maybe it should be a ModifierExtension then https://www.hl7.org/fhir/extensibility.html#modifierExtension.  If it wasn't that important, then would it be valid to have the client application determine that?

**Comment by declankieran** [comment]
@artronics @urchy @RNatarajanNHS 

There is also MedicationRequest.statusChangeDate, which isn't being mapped to anything in UKCore.  We had a discussion about this a few weeks ago and I removed the move over to UKCore, where I was adopting the method described here

http://build.fhir.org/versions.html#extensions

I've asked on Zulip and so far the consensus seems to be this is the method to adopt for this sort of thing

https://chat.fhir.org/#narrow/stream/179166-implementers/topic/Maintaining.20data.20across.20versions/near/275522171

The change was done here

https://github.com/NHSDigital/fhir-transforms/commit/c89d1eae2b9487c75c5ae213057f32295d9feaf9

When finding examples, I only found instances of MedicationRequest.statusChangeDate, I don't think there were any of dosageLastChanged, so my questions at this point are

1. Should I revert back to carrying over MedicationRequest.statusChangeDate.  I haven't yet looked into what happen here in the clincial and technical assurance, but it seems to be at least the same kind of thing as the dosageLastChanged scenario?
2. Should I manually create data for examples of dosageLastChanged as examples based on the definitions?  I think this needs done across the board anyway if you want an exhaustive list for your own test automation.
